src/vsr/checkpoint_trailer.zig

```diff
@@ -134,18 +129,19 @@ pub fn CheckpointTrailerType(comptime Storage: type) type {
 
             return .{
                 .trailer_type = trailer_type,
-                .write_block = write_block,
-                .buffer = buffer,
+                .blocks = blocks,
+                .block_bodies = block_bodies,
                 .block_addresses = block_addresses,
                 .block_checksums = block_checksums,
             };
         }
 
         pub fn deinit(trailer: *Self, allocator: mem.Allocator) void {
+            for (trailer.blocks) |block| allocator.free(block);
```

Comment:
Move this right before `allocator.free(trailer.blocks);` to keep reverse allocation order.